File: helpers/randomImage.js
var XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;

var getImage = function getImage(){
  valid = false;
  url = '';

  while(!valid) {
    var rstring = Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 5);
    var http = new XMLHttpRequest();
    http.open('HEAD', 'http://i.imgur.com/' + rstring + '.jpg', false);
    http.send();
    
    if (http.status != 404) {
      valid = true;
      url = 'http://i.imgur.com/' + rstring + '.jpg';
    }
  }
  
  return url;
};

module.exports.getImage = getImage;

File: helpers/randomImage.py
import random
import string
import requests

def getImage():
    valid = False
    code = ''

    while not valid:
        code = ''.join(random.choices(string.ascii_uppercase + string.digits, k=5))
        response = requests.get('http://i.imgur.com/' + code + '.jpg')
        if response.status_code == 200:
            if response.headers['Content-Length'] != '503':
                valid = True

    return 'http://i.imgur.com/' + code + '.jpg'